### DISCUSSION TITLE
Lauguerre-Gaussian Beam

### DISCUSSION THREAD

**Discussion by phy-ak** [epochpic/epoch#586] [Q&A]
1.) How to define a Laguerre-Gaussian laser beam which carry well-defined orbital angular momentum (OAM) along the laser propagation? Something like laser beam
consists of a hollow intensity distribution and helical wavefronts, often called twisted light.
2.) how to set the beam waist at 2.4 μm in the OAM case and 4.8 μm
in the Gaussian case?

**Comment by Status-Mirror** [comment]
Hey @Arun-lime,

I haven't worked with Laguerre-Gaussian beams before, but you may find the discussion in our [Gaussian beam demo](https://epochpic.github.io/quickstart/basic_examples/laser.html#focussing_a_gaussian_beam) helpful. In general, you'll want to find a functional form of your beam, deduce the values of the $E_y$ and $E_z$ fields on the $x_\text{min}$ boundary, and then set the profile and phase at this boundary to match your beam. Then, providing you use a beam which satisfies Maxwell's equations, the beam should propagate out as expected.

Cheers,
Stuart

**Reply by ghost** [reply]
Hello @Status-Mirror,

a question regarding the Gaussian beam in 3D. The demo block provided above is very useful but it is for 2D. What changes are necessary for making the laser in 3d? I have tried with putting r instead of y at phase and profile and also define at constants r=sqrt(y^2+z^2). Still it does not work properly so maybe I am making other mistakes.

What do you think?

Thank  you, in advance.

**Reply by Status-Mirror** [reply]
Hey @zj-cm,

I would have thought that this would be enough to convert to 3d. Can you copy in your input.deck below? If you put \``` above and below, you'll get a good formatting

\```
This
\```

becomes

```
This
```

Cheers,
Stuart

**Reply by ghost** [reply]
@Status-Mirror, here is an example of an input.deck (we are also trying to increase the computational power in order to have at least around 20 grid points/lambda on Ox). This kind of input.deck is also given in the user manual for 2D, so nothing so special, only I tried to make changes for 3D. 

Thank you.

```
begin : constant
lambda0=0.8*micron
temp=0
c_l=2.99792458e8
D_0 = 25*micron
w_0 = D_0 / 2.0
x_spot = 3.75e-6
Z_R = (pi * (w_0)^2) / lambda0
W_x = (w_0) * sqrt(1 + (x_spot/ Z_R)^2)
R_c = x_spot * (1 + (Z_R / x_spot)^2)
g_phase = atan(x_spot / Z_R)
r=sqrt(y^2+z^2)
end : constant

begin : control
nx = 1500
ny = 170
nz=ny
stdout_frequency=100
t_end = 600 * femto
x_min = 0.0
x_max = 150.0 * lambda0
y_min = - 32.0 * lambda0
y_max = 32.0 * lambda0
z_min = y_min
z_max = y_max
end : control


begin : boundaries
bc_x_min = simple_laser
bc_x_max = simple_outflow
bc_y_min = periodic
bc_y_max = periodic
bc_z_min = periodic
bc_z_max = periodic
end : boundaries

begin : laser
boundary = x_min
intensity_w_cm2 = 4e20
lambda = lambda0
pol = 0.0
phase = (2.0 * pi/lambda0)*(r^2/(2.0*R_c)) - g_phase
profile = gauss (r, 0, W_x)
t_profile = gauss (time, 50* femto , 25* femto)
end : laser

begin:species
name = electron
charge = - 1.0
mass = 1.0
frac = 1.0
temp = 0
density = 1e24
npart_per_cell = 4
end:species

begin:species
name = ions
charge = 2.0
mass = 4.0 * 1836
npart_per_cell = 2
immobile = T
density = 0.5e24
temp = 0.0
end:species

begin : window
move_window = T
window_v_x =  c_l
window_start_time = (x_max - x_min)/c_l
bc_x_min_after_move = simple_outflow
bc_x_max_after_move = simple_outflow
end : window

begin : output
dt_snapshot = 50 * femto
grid = always
ex = always
ey = always
ez = always
bx = always
by = always
bz = always
particles = always
px = always
py = always
pz = always
average_particle_energy = always + species
number_density = always + species
particle_energy_flux = always + species
mass_density = never + species
charge_density = always + species
temperature = never
end : output

```
